bazel/packaging/packaging.bzl

```diff
@@ -128,34 +131,130 @@ def _prepare_package_content(ctx, dynamic_loader_path = "/opt/redpanda/lib"):
+    cfg["directory_mode"] = True
+    if ctx.file.default_yaml_config != None:
+        cfg["package_files"].append({
+            "path": "config",
+            "name": "redpanda.yaml",
+            "source": ctx.file.default_yaml_config.path,
+        })
+    return cfg
+
+def _tarball_package_configuration(ctx, package_content, fips_enabled):
+    cfg = _common_redpanda_package_cfg(ctx, package_content, fips_enabled, "opt/redpanda")
+    cfg["binaries"].append({"path": "opt/redpanda/bin", "name": "redpanda", "source": package_content.redpanda_binary.path})
+    cfg["directory_mode"] = False
+    cfg["package_dirs"].append("etc/redpanda")
+    cfg["package_dirs"].append("var/lib/redpanda/data")
+    if ctx.file.default_yaml_config != None:
+        cfg["package_files"].append({
+            "path": "/etc/redpanda",
+            "name": "redpanda.yaml",
+            "source": ctx.file.default_yaml_config.path,
+        })
+    return cfg
+
 def _impl(ctx):
     use_dir = not ctx.attr.out.endswith(".tar.gz")
     out = ctx.actions.declare_directory(ctx.attr.out) if use_dir else ctx.actions.declare_file(ctx.attr.out)
-    package_content = _prepare_package_content(ctx)
+    package_content = _prepare_redpanda_package_content(ctx)
 
     fips_enabled = ctx.file.fips_module != None
     if fips_enabled != (ctx.file.fips_config != None):
         fail("`fips_module` and `fips_config` must both be specified in", ctx.attr.name)
+    if use_dir:
+        cfg = _dir_package_configuration(ctx, package_content, fips_enabled)
+    else:
+        cfg = _tarball_package_configuration(ctx, package_content, fips_enabled)
 
     # Create the configuration file for the packaging tool
     cfg_file = ctx.actions.declare_file("%s.config.json" % ctx.attr.name)
-    cfg = {
-        "redpanda_binary": package_content.redpanda_binary.path,
-        "rp_util": package_content.rp_util.path if package_content.rp_util else None,
-        "iotune": package_content.iotune.path if package_content.iotune else None,
-        "hwloc_calc": package_content.hwloc_calc.path if package_content.hwloc_calc else None,
-        "hwloc_distrib": package_content.hwloc_distrib.path if package_content.hwloc_distrib else None,
-        "openssl": package_content.openssl.path if package_content.openssl else None,
-        "rpk": package_content.rpk_binary.path if package_content.rpk_binary else None,
-        "shared_libraries": [solib.path for solib in package_content.shared_libraries],
-        "default_yaml_config": ctx.file.default_yaml_config.path if ctx.file.default_yaml_config else None,
-        "fips": {"module": ctx.file.fips_module.path, "config": ctx.file.fips_config.path} if fips_enabled else None,
-        "owner": ctx.attr.owner,
-        "directory_mode": use_dir,
-    }
     ctx.actions.write(cfg_file, content = json.encode_indent(cfg))
 
     inputs = [cfg_file, package_content.redpanda_binary] + package_content.shared_libraries
```

Comment:
should everything become optional?

Reply:
well, not really. My idea is to create a separate rule maybe something like `native_package` to define those generic package. Redpanda one is very specialised and i think it will have to stay that way with all the fips, redpanda.yaml etc